### PR TITLE
Feat/keyboard dropdown menu

### PR DESCRIPTION
**Pull request by TaiyeI**
Non functional dropdown menu to select Keyboard layout
image

### CODE REVIEW

File: src/components/KeyboardLayout/KeyboardSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Select from "react-select";
+import { KBLayout } from "../../enums/KBLayout";
+
+const keys = Object.keys(KBLayout);
+
+var options: { value: string; label: string }[] = [];
+
+keys.map((key) => {
+  options.push({ value: key, label: key });
+});
```
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: Can you create keys with the mapped array (merge the instantiation + push)

File: src/components/KeyboardLayout/KeyboardSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Select from "react-select";
+import { KBLayout } from "../../enums/KBLayout";
+
+const keys = Object.keys(KBLayout);
```
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: Can you move this inside the component?
We don't generally want variables outside of the component

**Review by KGDavidson — CHANGES_REQUESTED** [review]
A few changes needed here.
Also, please include a screenshot of UI changes in PR desc as we have discussed (classic coworker pasag)

File: package-lock.json
Comment (KGDavidson) [inline package-lock.json:1]: Remove this file. We use `yarn` not `npm`

File: src/components/KeyboardLayout/KeyboardSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { Dispatch, SetStateAction } from "react";
+
+type Props = {
+  setLayout: Dispatch<SetStateAction<KBLayoutType>>;
+};
+
+export const KeyboardSelect = (props: Props) => {
+  const { setLayout } = props;
+
+  const options: { value: string; label: string }[] = Object.entries(
```
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: Implicit typing is enough here, no need for overdefining

File: src/components/KeyboardLayout/KeyboardSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  return (
+    <div>
+      <Select
+        classNames={{
+          control: () => "!bg-zinc-700 !border-zinc-700 !shadow-none",
+          singleValue: () => "!text-zinc-400",
+          menu: () => "!bg-zinc-700 !text-zinc-400",
+          option: () => "hover:bg-zinc-600 bg-transparent active:bg-zinc-600",
```
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: Can we reduce the use of !important here? (we can)

File: src/components/KeyboardLayout/KeyboardSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const { setLayout } = props;
+
+  const options: { value: string; label: string }[] = Object.entries(
+    KBLayout
+  ).map((keyValue) => {
+    return { value: keyValue[1], label: keyValue[0] };
+  });
+
+  return (
+    <div>
```
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: Remove this extra div

File: src/components/KeyboardLayout/KeyboardSelect.tsx (hunk 3 above)
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: I don't like that we are recomputing this every time the component renders. Can we memoise it?

**Review by KGDavidson — CHANGES_REQUESTED** [review]
A few things need to be updated :) 
Also please include a full page screenshot so that we can see the component in context as is

File: src/components/KeyboardLayout/KeyboardSelect.tsx (hunk 4 above)
Comment (TaiyeI) [reply src/components/KeyboardLayout/KeyboardSelect.tsx]: Had to add another one for bg-transparent

File: src/components/KeyboardLayout/KeyboardSelect.tsx [hunk 6]
```diff
@@ -9,27 +9,26 @@ type Props = {
...
   const { setLayout } = props;
 
-  const options: { value: string; label: string }[] = Object.entries(
-    KBLayout
-  ).map((keyValue) => {
+  const options = Object.entries(KBLayout).map((keyValue) => {
     return { value: keyValue[1], label: keyValue[0] };
   });
 
+  const Exoptions = useMemo(() => options, [KBLayout]);
```
Comment (KGDavidson) [inline src/components/KeyboardLayout/KeyboardSelect.tsx]: bad

**Review by KGDavidson — DISMISSED** [review]
lgtm

**Review by KGDavidson — APPROVED** [review]

**Pull request merged**